qcodes/dataset/sqlite_base.py

```diff
@@ -1904,6 +1911,39 @@ def get_metadata(conn: SomeConnection, tag: str, table_name: str):
                             "result_table_name", table_name)
 
 
+def get_metadata_from_run_id(conn: SomeConnection, run_id: int) -> Dict:
+    """
+    Get all metadata associated with the specified run
+    """
+    # TODO: promote snapshot to be present at creation time
+    non_metadata = RUNS_TABLE_COLUMNS + ['snapshot']
+
+    metadata = {}
+    possible_tags = []
+
+    # first fetch all columns of the runs table
+    query = "PRAGMA table_info(runs)"
+    cursor = conn.cursor()
+    for row in cursor.execute(query):
+        if row['name'] not in non_metadata:
+            possible_tags.append(row['name'])
+
+    # and then fetch whatever metadata the run might have
+    for tag in possible_tags:
+        query = f"""
+                SELECT "{tag}"
+                FROM runs
+                WHERE run_id = ?
+                AND "{tag}" IS NOT NULL
```

Comment:
i was trying to find when this extra condition would give problems, but couldn't.